parsec/mca/device/device_gpu.c

```diff
@@ -1510,11 +1524,14 @@ parsec_device_data_stage_in( parsec_device_gpu_module_t* gpu_device,
 
     /* update the data version in GPU immediately, and mark the data under transfer */
     assert((gpu_elem->version != candidate->version) || (gpu_elem->data_transfer_status == PARSEC_DATA_STATUS_NOT_TRANSFER));
-    gpu_elem->version = candidate->version;
+    if( PARSEC_FLOW_ACCESS_WRITE & type && gpu_task->task_type != PARSEC_GPU_TASK_TYPE_PREFETCH )
+        gpu_elem->version = candidate->version + 1;
+    else
+        gpu_elem->version = candidate->version;
```

Comment:
Can you please document why we set the version the way we set it? This is delicate and not obvious.